cmd/scip/test_test.go

```diff
@@ -0,0 +1,208 @@
+	require.Equal(t, "reference", actual[0].attribute.kind)
+	require.Equal(t, "reference", actual[1].attribute.kind)
+	require.Equal(t, "reference", actual[2].attribute.kind)
+	require.Equal(t, "diagnostic", actual[3].attribute.kind)
+
+	// validate that additionalData is correctly parsed
+	require.True(
+		t,
+		isEqual(
+			actual[3].attribute.additionalData,
+			[]string{"with multiline", "additional data"},
+		),
+	)
+}
+
+func TestIsValidTestCaseForAttribute_length(t *testing.T) {
+	equalLength := isValidTestCaseForAttribute(
+		&symbolAttributeTestCase{
+			enforceLength: true,
+			attribute:     &symbolAttribute{start: 3, length: 5},
+		},
+		&symbolAttribute{start: 3, length: 5},
+	)
+	require.True(t, equalLength)
+
+	notEqualLength := isValidTestCaseForAttribute(
+		&symbolAttributeTestCase{
+			enforceLength: true,
+			attribute:     &symbolAttribute{start: 3, length: 5},
+		},
+		&symbolAttribute{start: 3, length: 4},
+	)
+	require.False(t, notEqualLength)
+}
+
+func TestIsValidTestCaseForAttribute_start(t *testing.T) {
+	validStart := isValidTestCaseForAttribute(
+		&symbolAttributeTestCase{
+			attribute: &symbolAttribute{start: 4},
+		},
+		&symbolAttribute{start: 3, length: 5},
+	)
+	require.True(t, validStart)
+
+	invalidStart := isValidTestCaseForAttribute(
+		&symbolAttributeTestCase{
+			attribute: &symbolAttribute{start: 9},
+		},
+		&symbolAttribute{start: 3, length: 5},
+	)
+	require.False(t, invalidStart)
+}
+
+func TestIsValidTestCaseForAttribute_kind(t *testing.T) {
+	equalKind := isValidTestCaseForAttribute(
+		&symbolAttributeTestCase{
+			attribute: &symbolAttribute{start: 4, kind: "definition"},
+		},
+		&symbolAttribute{start: 3, length: 5, kind: "definition"},
+	)
+	require.True(t, equalKind)
+
+	notEqualKind := isValidTestCaseForAttribute(
+		&symbolAttributeTestCase{
+			attribute: &symbolAttribute{start: 4, kind: "reference"},
+		},
+		&symbolAttribute{start: 3, length: 5, kind: "definition"},
+	)
+	require.False(t, notEqualKind)
+}
+
+func TestIsValidTestCaseForAttribute_data(t *testing.T) {
+	equalSymbol := isValidTestCaseForAttribute(
+		&symbolAttributeTestCase{
+			attribute: &symbolAttribute{
+				start: 4,
+				kind:  "definition",
+				data:  "scip-dart pub somePackage 1.0.0 lib/`foo.dart`/main().",
+			},
+		},
+		&symbolAttribute{
+			start:  3,
+			length: 5,
+			kind:   "definition",
+			data:   "scip-dart pub somePackage 1.0.0 lib/`foo.dart`/main().",
+		},
+	)
+	require.True(t, equalSymbol)
+
+	notEqualSymbol := isValidTestCaseForAttribute(
+		&symbolAttributeTestCase{
+			attribute: &symbolAttribute{
+				start: 4,
+				kind:  "definition",
+				data:  "scip-dart pub somePackage 1.0.0 lib/`foo.dart`/main().",
+			},
+		},
+		&symbolAttribute{
+			start:  3,
+			length: 5,
+			kind:   "definition",
+			data:   "scip-dart pub somePackage 1.0.0 lib/`foo.dart`/bar().",
+		},
+	)
+	require.False(t, notEqualSymbol)
+}
+
+func TestIsValidTestCaseForAttribute_additionalData(t *testing.T) {
```

Comment:
Can we reuse the code we have for reprolang (to easily generate a SCIP index with the desired data) as well as the test parsing here (to generate the test cases) so that the bulk of tests here involve end-to-end testing and avoiding having to write a bunch of struct literals by hand?